src/S_Object/S_Cone.cpp

```diff
@@ -0,0 +1,69 @@
+#include <sch/S_Object/S_Cone.h>
+
+using namespace sch;
+
+const Point3 S_Cone::origin_(0,0,0);
+
+S_Cone::S_Cone(const Scalar& angle, const Scalar &  height) 
+
+{
+
+  angle_=angle;
```

Comment:
I think it would make sense to check that `angle < M_PI/2` and throw `std::invalid_argument` otherwise

Reply:
Good practice indeed, we need to extend it to other objects

Reply:
Should be rather std::domain_error though, no?

Reply:
Solved with https://github.com/mehdi-benallegue/sch-core/commit/c91a14a99e32baff748f9468e2490ad40d8fa867